applications/vanilla/modules/class.newdiscussionmodule.php

```diff
@@ -72,7 +72,7 @@ public function addButton($Text, $Url) {
      */
     public function toString() {
         // Set CategoryID if we have one.
-        if ($this->CategoryID === null) {
+        if ($this->CategoryID === null && c('Vanilla.Categories.Use', true)) {
```

Comment:
Switch up the config check to be the first condition.  If that fails, there's no point in checking the CategoryID.